e2e/cypress/integration/team/remove_team_icon_spec.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+let team;
+
+function openTeamSettingsDialog() {
+    // validating the side bar is visible
+    cy.get('#sidebarHeaderDropdownButton').should('be.visible');
+
+    // clicking on the side bar
+    cy.get('#sidebarHeaderDropdownButton').click();
+
+    // Team settings button must be visible
+    cy.get('#teamSettings').should('be.visible').should('contain', 'Team Settings');
+
+    //click on the Team settings button
+    cy.get('#teamSettings').click();
+
+    // verify the team settings dialog is open
+    cy.get('#teamSettingsModalLabel').should('be.visible').should('contain', 'Team Settings');
+
+    // verify the edit icon is visible
+    cy.get('#team_iconEdit').should('be.visible');
+
+    // clicking on edit button
+    cy.get('#team_iconEdit').click();
+}
+
+describe('Teams Suite', () => {
+    before(() => {
+        cy.apiUpdateConfig({EmailSettings: {RequireEmailVerification: false}});
+
+        // # Login as new user
+        cy.loginAsNewUser().then(() => {
+            // # Create new team and visit its URL
+            cy.apiCreateTeam('test-team', 'Test Team').then((response) => {
+                team = response.body;
+                cy.visit(`/${team.name}`);
+            });
+        });
+    });
+
+    it('TS14632 Remove Team Icon', () => {
+        // function to open the team settings dialog
+        openTeamSettingsDialog();
+
+        // verify the settings picture button is visible to click
+        cy.get('[data-testid="inputSettingPictureButton"]').should('be.visible').click();
```

Comment:
Hi @saturninoabril, thanks for the suggestion but as per @reflog review comments I have followed the https://docs.cypress.io/guides/references/best-practices.html and update the _ids_ to _data-testid_
Looking at the existing code base I can see the mix approach few of the specs has cy.findByTestId and few of them are using cy.get('[data-testid="xyz"]'). 
It would be really better if the approaches are consistent across, but please advise which approach is better?

Reply:
It's not a contrary to his comment.  Since you're using `data-testid`, the convenience function is to get it by `cy.findByTestId`.  Those using ``cy.get('[data-testid="xyz"]')`` works too but since we have ``cy.findByTestId``, let's use it.
We can fix others in separate PR to make it consistent all throughout the codebase.